app/src/main/java/album2/yassinelhadedy/github/com/albumapp2/MainActivity.kt

```diff
@@ -0,0 +1,82 @@
+package album2.yassinelhadedy.github.com.albumapp2
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.Toolbar
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import retrofit2.Call
+import retrofit2.Response
+import retrofitrx.yassinelhadedy.github.com.albumapp.MainAdapter
+import retrofitrx.yassinelhadedy.github.com.albumapp.api.PhotoReciver
+import retrofitrx.yassinelhadedy.github.com.albumapp.model.Photo
+import retrofitrx.yassinelhadedy.github.com.albumapp.model.PhotoList
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
+    override fun onClick(view: View?) {
+        val intent= Intent(this,Details::class.java)
+        val holder =view?.tag as MainAdapter.PhotoViewHolder
+        intent.putExtra(Details.PHOTO,mainadapter?.getphoto(holder.adapterPosition))
+        startActivity(intent)
+    }
+
+    var photos:List<Photo>?=null
+
+    var mainadapter: MainAdapter?=null
```

Comment:
- mainadapter -> mainAdapter
- Try to find a delegate so that this variable does not allow nulls.